apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/chainid/HeightConfig.java

```diff
@@ -20,12 +20,10 @@
     private final long maxBaseTarget;
     private final long minBaseTarget;
 
-    public HeightConfig(BlockchainProperties bp) {
-        Objects.requireNonNull(bp, "Blockchain properties cannot be null");
-
-        this.bp = bp;
+    public HeightConfig(BlockchainProperties bp, long oneAPL) {
+        this.bp = Objects.requireNonNull(bp, "Blockchain properties cannot be null");
         this.maxPayloadLength = bp.getMaxNumberOfTransactions() * Constants.MIN_TRANSACTION_SIZE;
-        this.maxBalanceAtm = bp.getMaxBalance() * Constants.ONE_APL;
+        this.maxBalanceAtm = bp.getMaxBalance() * oneAPL;
```

Comment:
Use Math.multiplyExact to avoid overflow.

Reply:
fixed